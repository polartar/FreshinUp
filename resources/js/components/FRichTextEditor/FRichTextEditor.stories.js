import { storiesOf } from '@storybook/vue'
// import { action } from '@storybook/addon-actions'

import FRichTextEditor from './FRichTextEditor'

const sampleText = `
<h2>
  Hi there,
</h2>
<p>
  this is a very <em>basic</em> example of tiptap.
</p>
<pre><code>body { display: none; }</code></pre>
<ul>
  <li>
    A regular list
  </li>
  <li>
    With regular items
  </li>
</ul>
<blockquote>
  It's amazing üëè
  <br />
  ‚Äì mom
</blockquote>
`

export const Default = () => ({
  components: { FRichTextEditor },
  template: `
    <v-container>
      <f-rich-text-editor />
    </v-container>
  `
})

export const Populated = () => ({
  components: { FRichTextEditor },
  data () {
    return {
      text: sampleText
    }
  },
  template: `
    <v-container>
      <f-rich-text-editor v-model="text" />
    </v-container>
  `
})

storiesOf('FoodFleet|components/FRichTextEditor', module)
  .addParameters({
    backgrounds: [
      { name: 'default', value: '#f1f3f6', default: true }
    ]
  })
  .add('Default', Default)
  .add('Populated', Populated)
