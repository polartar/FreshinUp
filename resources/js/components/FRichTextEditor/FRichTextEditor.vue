<template>
  <div class="editor">
    <editor-menu-bar
      v-slot="{ commands, isActive }"
      :editor="editor"
    >
      <div class="menubar">
        <div class="menubar">
          <button
            class="menubar__button"
            :class="{ 'is-active': isActive.heading({ level: 1 }) }"
            @click="commands.heading({ level: 1 })"
          >
            H1
          </button>

          <button
            class="menubar__button"
            :class="{ 'is-active': isActive.heading({ level: 2 }) }"
            @click="commands.heading({ level: 2 })"
          >
            H2
          </button>

          <button
            class="menubar__button"
            :class="{ 'is-active': isActive.heading({ level: 3 }) }"
            @click="commands.heading({ level: 3 })"
          >
            H3
          </button>

          <v-btn
            icon
            depressed
            :class="{ 'is-active': isActive.bold() }"
            @click="commands.bold"
          >
            <v-icon>format_bold</v-icon>
          </v-btn>

          <v-btn
            icon
            depressed
            :class="{ 'is-active': isActive.italic() }"
            @click="commands.italic"
          >
            <v-icon>format_italic</v-icon>
          </v-btn>

          <v-btn
            icon
            depressed
            :class="{ 'is-active': isActive.underline() }"
            @click="commands.underline"
          >
            <v-icon>format_underlined</v-icon>
          </v-btn>

          <v-btn
            icon
            depressed
            :class="{ 'is-active': editor.activeMarkAttrs.aligntext.align === 'left' }"
            @click="commands.aligntext({ align: 'left' })"
          >
            <v-icon>format_align_left</v-icon>
          </v-btn>

          <v-btn
            icon
            depressed
            :class="{ 'is-active': editor.activeMarkAttrs.aligntext.align === 'center' }"
            @click="commands.aligntext({ align: 'center' })"
          >
            <v-icon>format_align_center</v-icon>
          </v-btn>

          <v-btn
            icon
            depressed
            :class="{ 'is-active': editor.activeMarkAttrs.aligntext.align === 'right' }"
            @click="commands.aligntext({ align: 'right' })"
          >
            <v-icon>format_align_right</v-icon>
          </v-btn>

          <v-btn
            icon
            depressed
            :class="{ 'is-active': editor.activeMarkAttrs.aligntext.align === 'justify' }"
            @click="commands.aligntext({ align: 'justify' })"
          >
            <v-icon>format_align_justify</v-icon>
          </v-btn>

          <v-btn depressed>
            Add variable
          </v-btn>
        </div>
      </div>
    </editor-menu-bar>
    <editor-content
      class="editor__content pa-3"
      :editor="editor"
    />
  </div>
</template>
<script>
import { Editor, EditorContent, EditorMenuBar } from 'tiptap'
import AlignText from '../tiptap/aligntext.plugin'
import {
  // Blockquote,
  // CodeBlock,
  // HardBreak,
  Heading,
  // HorizontalRule,
  // OrderedList,
  // BulletList,
  // ListItem,
  // TodoItem,
  // TodoList,
  Bold,
  // Code,
  Italic,
  // Link,
  // Strike,
  Underline
  // History
} from 'tiptap-extensions'

export default {
  components: {
    EditorContent,
    EditorMenuBar
  },

  props: {
    value: {
      type: String,
      default: ''
    }
  },

  data: () => ({
    editor: null
  }),

  watch: {
    value (val) {
      if (this.editor && val !== this.value) {
        this.editor.setContent(val, true)
      }
    }
  },

  mounted: function () {
    this.$nextTick(() => this.init())
  },

  beforeDestroy: function () {
    if (this.editor) {
      this.editor.destroy()
    }
  },

  methods: {
    async init () {
      this.editor = new Editor({
        extensions: [
          new Heading(),
          new Bold(),
          new Italic(),
          new Underline(),
          new AlignText()
        ],
        content: this.value,
        onUpdate: ({ getHTML }) => {
          const state = getHTML()
          this.$emit('input', state)
        }
      })
    }
  }
}
</script>
<style scoped>
.editor {
  position: relative;
  /* max-width: 30rem; */
  margin: 0 auto 5rem;
}

.menubar {
  margin-bottom: 1rem;
  -webkit-transition: visibility .2s .4s,opacity .2s .4s;
  transition: visibility .2s .4s,opacity .2s .4s;
}

.editor__content {
  overflow-wrap: break-word;
  word-wrap: break-word;
  word-break: break-word;
  border: solid 2px darkgray;
  border-radius: 4px;
}

.menubar__button {
  font-weight: 700;
  display: -webkit-inline-box;
  display: inline-flex;
  background: transparent;
  border: 0;
  color: #000;
  padding: .2rem .5rem;
  margin-right: .2rem;
  border-radius: 3px;
  cursor: pointer;
}
>>>.ProseMirror {
  outline: none;
}
</style>
