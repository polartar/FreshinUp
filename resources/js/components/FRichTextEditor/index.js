import FRichTextEditor from './FRichTextEditor'
export default FRichTextEditor
