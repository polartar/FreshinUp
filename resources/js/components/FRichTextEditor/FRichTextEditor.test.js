import * as Stories from './FRichTextEditor.stories'
import Component from './FRichTextEditor'
import { mount, shallowMount } from '@vue/test-utils'

describe('component/FRichTextEditor', () => {
  describe('Snapshots', () => {
    test('Default', async () => {
      const wrapper = mount(Stories.Default())
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
    test('Populated', async () => {
      const wrapper = mount(Stories.Populated())
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Props', () => {
    test('value', async () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.value).toBe('')

      const template = Stories.SAMPLE_TEXT
      wrapper.setProps({
        value: template
      })
      expect(wrapper.vm.value).toBe(template)
    })
  })
})
