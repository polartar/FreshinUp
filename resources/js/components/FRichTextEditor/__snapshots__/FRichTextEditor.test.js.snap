// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/FRichTextEditor Snapshots Default 1`] = `
<div
  class="container"
>
  <div
    class="editor"
  >
    <div
      class="menubar"
    >
      <div
        class="menubar"
      >
        <button
          class="menubar__button"
        >
          
          H1
        
        </button>
         
        <button
          class="menubar__button"
        >
          
          H2
        
        </button>
         
        <button
          class="menubar__button"
        >
          
          H3
        
        </button>
         
        <button
          class="v-btn v-btn--icon v-btn--depressed theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              format_bold
            </i>
          </div>
        </button>
         
        <button
          class="v-btn v-btn--icon v-btn--depressed theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              format_italic
            </i>
          </div>
        </button>
         
        <button
          class="v-btn v-btn--icon v-btn--depressed theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              format_underlined
            </i>
          </div>
        </button>
         
        <button
          class="v-btn v-btn--icon v-btn--depressed theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              format_align_left
            </i>
          </div>
        </button>
         
        <button
          class="v-btn v-btn--icon v-btn--depressed theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              format_align_center
            </i>
          </div>
        </button>
         
        <button
          class="v-btn v-btn--icon v-btn--depressed theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              format_align_right
            </i>
          </div>
        </button>
         
        <button
          class="v-btn v-btn--icon v-btn--depressed theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              format_align_justify
            </i>
          </div>
        </button>
         
        <button
          class="v-btn v-btn--depressed theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            
          Add variable
        
          </div>
        </button>
      </div>
    </div>
     
    <div
      class="editor__content pa-3"
    >
      <div
        class="ProseMirror"
        contenteditable="true"
        tabindex="0"
      >
        <p>
          <br />
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`component/FRichTextEditor Snapshots Populated 1`] = `
<div
  class="container"
>
  <div
    class="editor"
  >
    <div
      class="menubar"
    >
      <div
        class="menubar"
      >
        <button
          class="menubar__button"
        >
          
          H1
        
        </button>
         
        <button
          class="menubar__button is-active"
        >
          
          H2
        
        </button>
         
        <button
          class="menubar__button"
        >
          
          H3
        
        </button>
         
        <button
          class="v-btn v-btn--icon v-btn--depressed theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              format_bold
            </i>
          </div>
        </button>
         
        <button
          class="v-btn v-btn--icon v-btn--depressed theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              format_italic
            </i>
          </div>
        </button>
         
        <button
          class="v-btn v-btn--icon v-btn--depressed theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              format_underlined
            </i>
          </div>
        </button>
         
        <button
          class="v-btn v-btn--icon v-btn--depressed theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              format_align_left
            </i>
          </div>
        </button>
         
        <button
          class="v-btn v-btn--icon v-btn--depressed theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              format_align_center
            </i>
          </div>
        </button>
         
        <button
          class="v-btn v-btn--icon v-btn--depressed theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              format_align_right
            </i>
          </div>
        </button>
         
        <button
          class="v-btn v-btn--icon v-btn--depressed theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              format_align_justify
            </i>
          </div>
        </button>
         
        <button
          class="v-btn v-btn--depressed theme--light"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            
          Add variable
        
          </div>
        </button>
      </div>
    </div>
     
    <div
      class="editor__content pa-3"
    >
      <div
        class="ProseMirror"
        contenteditable="true"
        tabindex="0"
      >
        <h2>
          Hi there,
        </h2>
        <p>
          this is a very 
          <em>
            basic
          </em>
           example of tiptap.
        </p>
        <p>
          body { display: none; }
        </p>
        <p>
           A regular list
        </p>
        <p>
           With regular items
        </p>
        <p>
           It's amazing üëè ‚Äì mom
        </p>
      </div>
    </div>
  </div>
</div>
`;
